### DISCUSSION TITLE
Cannot set heating or charging

### DISCUSSION THREAD

**Discussion by entsukki** [Hyundai-Kia-Connect/kia_uvo#376] [Q&A] [answered]
Hi
I have e-niro 2021 ev (EU). My issue is that i have 34 entities showing. Mostly sensor data byt the only command i can automate is lock/unlock doors. What I am missing? I can set the charging times and use ac/heating with the uvo app no prob. But the services to start heating/ac (or schedule charge)cannot be found. I am not yet very familiar with HA so this might be some rookie question. But what should I do ? I thought these things should be supported?

**Comment by entsukki** [comment, marked as answer]
I just decided to to try one more time. And now I figured it out 🫣

**Reply by AceMoneus** [reply]
Hi! How did U do it?
I suppose I want the same thing as You. I want to set a button in HA and just press it and then the cars starts heating/cooling.
But I can't figure out how to do it... :(
I have a Kona from 2021, but I assume it's the same commands as KIA.

**Reply by cdnninja** [reply]
This is a service call.  Can access via the developer menu or an automation.

**Reply by AceMoneus** [reply]
Ah! Ok, thx. I'll dig in there.

**Reply by vahaldor** [reply]
I have a new sorento and these services do not work for me. In fact, starting the car or heating is not even available in the uvo app. Do I have to buy some device to the car in order to have these features available?

**Reply by cdnninja** [reply]
> I have a new sorento and these services do not work for me. In fact, starting the car or heating is not even available in the uvo app. Do I have to buy some device to the car in order to have these features available?
> 

 If not in app it won't work.   Either your trim doesn't support it or your region doesn't.

**Comment by carlosvalin** [comment]
How do I add the function to activate or deactivate vehicle charging?

**Reply by cdnninja** [reply]
Does it work in app? If not we can't do anything.  If it is in app go to the developer services menu.

**Reply by carlosvalin** [reply]
It works in the application, and from HA services , Is it possible to add a button for these actions like Door Lock?

**Reply by cdnninja** [reply]
> It works in the application, and from HA services , Is it possible to add a button for these actions like Door Lock?

Yes you can use automations tied to a push button in the dashboard if you like.   

Reason it's not exposed by default is settings have to be passed with it and no good way exists for that.

**Reply by reivilob** [reply]
sorry if somebody has already be able to add a button to start/stop charge i would appreciate some help because i am totally unable to do it ... 
(all other part of the Hundai-kia integration is working fine ...)
